sips/sip-010/sip-010-fungible-token-standard.md

```diff
@@ -0,0 +1,132 @@
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+## Trait implementation
+
+An implementation of the proposed trait is provided below.
+
+```clarity
+(define-trait ft-trait
+  (
+    ;; Transfer from the caller to a new principal
+    (transfer (principal uint) (response bool uint))
+
+    ;; the human readable name of the token
+    (name () (response (string-ascii 32) uint))
+
+    ;; the ticker symbol, or empty if none
+    (symbol () (response (string-ascii 32) uint))
+
+    ;; the number of decimals used, e.g. 6 would mean 1_000_000 represents 1 token
+    (decimals () (response uint uint))
+
+    ;; the balance of the passed principal
+    (balance-of (principal) (response uint uint))
+
+    ;; the current total supply (which does not need to be a constant)
+    (total-supply () (response uint uint))
+  )
+)
+```
+
+## Example implementation
+
+I have provided a simple implementation along with a Javascript client and tests. https://github.com/hstove/stacks-fungible-token
+
+## Credit
+
+Credit for the work behind this proposal belongs to [@psq](https://github.com/psq), who wrote almost all of this code originally for use in his projects (such as [Flexr](https://github.com/psq/flexr)). I am simply re-packaging his existing code in a format that can be shared in this proposal.
+
+## Implementing in wallets and other applications
+
+Developers who wish to interact with a fungible token contract should first be provided, or keep track of, various different fungible token implementations. When validating a fungible token contract, they should fetch the interface and/or source code for that contract. If the contract implements the trait, then the wallet can use this standard's contract interface for making transfers and getting balances.
+
+### Use of post conditions
+
+In addition to fantastic built-in methods for fungible token contracts, the Stacks blockchain includes an innovation known as Post Conditions. By defining post conditions, users can create transactions that include pre-defined guarantees about what might happen in that contract.
+
+One such post condition could be "I will transfer exactly 100 of X token", where "X token" is referenced as a specific contract's fungible token. When wallets and applications implement the `transfer` method, they should _always_ use post conditions to specify that the user will transfer exactly the amount of tokens that they specify in the `amount` argument of the `transfer` function. Only in very specific circumstances should such a post condition not be included.
+
+## Related work
+
+- [@psq's trait and implementation](https://github.com/psq/flexr/blob/master/contracts/src20-trait.clar)
+- [@friedger's fungible token implementation](https://github.com/friedger/clarity-smart-contracts/blob/master/contracts/tokens/fungible-token.clar)
+- [Ethereum ERC20 standard](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/)
```

Comment:
Per SIP-000, you'll need to add the following sections at some point.  Just jotting it down here so we remember to get it done.

* An `Activation` section, which describes a checklist for determining whether or not this SIP gets ratified.  In this case, you'd want the trait to be deployed to the network at the very least, and you'd probably also want something like "5 different token implementations exist by this date" and "at least one token is listed on a major exchange by this date" or text to that effect.  Basically, the SIP steering committee needs a way to determine that this SIP is accepted by the greater Stacks community, or rejected.

* A `Backwards Compatibility` section.  Are there any existing systems that this token standard needs to be compatible with, and if so, how is this achieved?  It's fine to leave this section blank if there are no backwards compatibility concerns, but the SIP should still include the boilerplate "Not applicable" text if so.

* A `Reference Implementations` section.  If you have any Clarity trait code and implementation code, you would include links to them here.  I think it would be wise to have at least two reference implementations for this trait included as supplemental material, so future implementers can have something to look at.